docs/getting-started.md

```diff
@@ -3,33 +3,111 @@ layout: docs
+- `read_repository`: Necessary to read the Staticman site config
+- `write_repository`: Necessary to merge pull requests
 
-## Step 2. Create a configuration file
+### **If using GitHub to host the static site repo:**
 
-I will look for a `staticman.yml` file in the root of the repository, where various configuration parameters will be defined.
+#### **Option 1. Authenticate as a GitHub application**
 
-You can use the [sample config file](https://github.com/eduardoboucas/staticman/blob/master/staticman.sample.yml) as a starting point and check the [available configuration parameters](/docs/configuration) for more information.
+This is the recommended way to authenticate with GitHub. This method will give the most control over what Staticman can and can't access.
 
-## Step 3. Hook up your forms
+1. <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/creating-a-github-app" class="cta">Create a new GitHub application</a>. Ensure you use the following:
+  - Homepage: `"https://staticman.net/"`
+  - Webhook URL: `"{STATICMAN_BASE_URL}/v1/webhook"` - e.x. `"https://mystaticmaninstance.herokuapp.com/v1/webhook"`
+  - Contents: `Read & Write` - Necessary to read the Staticman site config
+  - Pull Requests: `Read & Write` - Necessary to merge pull requests
+  - Subscribe to `Pull request` events
+1. Generate a private key for the app and note it down along with your app ID.
+1. Install the app on your GitHub account. You can limit its access to only the repo from which you host your static site
 
-Forms should `POST` to:
+<br>
 
-`https://api.staticman.net/v2/entry/{GITHUB USERNAME}/{GITHUB REPOSITORY}/{BRANCH}/{PROPERTY (optional)}`
+#### **Option 2. Authenticate to GitHub using a personal access token on a bot**
+
+1. Register a new GitHub account to run your Staticman bot and create a <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token" class="cta">personal access token</a> for this new account.
+1. From your main GitHub account, <a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/inviting-collaborators-to-a-personal-repository" class="cta">send your bot a collaboration invite</a>.
+1. Return to this step once your Staticman instance is running and send a GET request to
+  `{STATICMAN_BASE_URL}/v3/connect/{GIT_PROVIDER_USERNAME}/{REPO}`
+  <br>For example:<br>
+  `https://staticmaninstance.herokuapp.com/v3/connect/eduardoboucas/staticman.net`
+1. You should get an `OK!` response and the bot should have accepted the collaboration invitation.
+
+<br>
+
+#### **Option 3. Authenticate to GitHub using a personal access token on your main account**
+
+This option is not recommended as it gives Staticman direct and complete access to your primary GitHub account. Simply, create a personal access token on your primary account.
```

Comment:
I think we should add more clarification here. How about, 
> This option is **not recommended** as it gives Staticman direct and complete access to your primary GitHub account. In the unfortunate incident where your Staticman instance is compromised, it will get full access to all your repositories, private and public alike as well as you account and may result in irreversible damage across repositories and possibly organizational repositories you are member of. If you still want to proceed, start by creating a personal access token on your primary account.

Reply:
maybe it is supposed to be `{GIT_PROVIDER/GIT_PROVIDER_USERNAME}` instead of `{GIT_PROVIDER_USERNAME}` ?